src/Providers/Models/DTO/ModelRequirements.php

```diff
@@ -88,6 +91,192 @@ public function getRequiredOptions(): array
         return $this->requiredOptions;
     }
 
+    /**
+     * Checks whether the given model metadata meets these requirements.
+     *
+     * @since 0.2.0
+     *
+     * @param ModelMetadata $metadata The model metadata to check against.
+     * @return bool True if the model meets all requirements, false otherwise.
+     */
+    public function areMetBy(ModelMetadata $metadata): bool
+    {
+        // Check if all required capabilities are supported
+        foreach ($this->requiredCapabilities as $requiredCapability) {
+            $supported = false;
+            foreach ($metadata->getSupportedCapabilities() as $supportedCapability) {
```

Comment:
Instead of a nested `foreach` loop, which can be problematic for performance, why not maintain the original approach of using lookup maps that's present in `ModelMetadata::meetsRequirements`?

You can call `getSupportedCapabilities()` and `getSupportedOptions()` once, in the beginning of this method and then create a `$capabilitiesMap` and an `$optionsMap` from them, just like today in `ModelMetadata`.

This way you can take over the more efficient code as is - which already reduces the risk of introducing new bugs.